assets/app/view/game/game_info.rb

```diff
@@ -225,7 +225,7 @@ def trains
               event_text << if index.zero?
                               event_name
                             else
-                              "#{event_name}(on #{ordinal(index + 1)} train)"
+                              "#{event_name}(on #{ordinal(train2.index + 1)} train)"
```

Comment:
The relative index will change as trains of that type are bought. It should be the train's absolute index that matters.